### PR TITLE
SDN-3687: Support gathering DBs for OVNK-Interconnect mode

### PR DESCRIPTION
**Pull request by tssurya**
We are adding support for IC in 4.14. See
https://issues.redhat.com/browse/SDN-3733
for more details.

This commit ensures we have support for
gathering logs from both modes. We will remove
the legacy version in future a.k.a `gather_ovn_kubernetes_data_legacy_mode` will just be deleted

```
drwxr-xr-x 2 surya surya 9.5M Jul 17 18:03 ovnk_database_store
-rw-r--r-- 1 surya surya 1.6M Jul 17 18:03 ovnk_database_store.tar.gz
[surya@hidden-temple network_logs]$ cd ovnk_database_store/
[surya@hidden-temple ovnk_database_store]$ ls -ltrh
total 9.5M
-rw-r--r-- 1 surya surya 1.4M Jul 17 18:03 ovnkube-node-ssh2b_sbdb
-rw-r--r-- 1 surya surya 483K Jul 17 18:03 ovnkube-node-ssh2b_nbdb
-rw-r--r-- 1 surya surya 1.4M Jul 17 18:03 ovnkube-node-s4597_sbdb
-rw-r--r-- 1 surya surya 508K Jul 17 18:03 ovnkube-node-s4597_nbdb
-rw-r--r-- 1 surya surya 1.1M Jul 17 18:03 ovnkube-node-r5srt_sbdb
-rw-r--r-- 1 surya surya 336K Jul 17 18:03 ovnkube-node-r5srt_nbdb
-rw-r--r-- 1 surya surya 868K Jul 17 18:03 ovnkube-node-qz4sq_sbdb
-rw-r--r-- 1 surya surya 272K Jul 17 18:03 ovnkube-node-qz4sq_nbdb
-rw-r--r-- 1 surya surya 1.5M Jul 17 18:03 ovnkube-node-8d8qp_sbdb
-rw-r--r-- 1 surya surya 554K Jul 17 18:03 ovnkube-node-8d8qp_nbdb
-rw-r--r-- 1 surya surya 979K Jul 17 18:03 ovnkube-node-564lr_sbdb
-rw-r--r-- 1 surya surya 314K Jul 17 18:03 ovnkube-node-564lr_nbdb
[surya@hidden-temple ovnk_database_store]$
```
```
[must-gather-lptfw] POD 2023-07-17T16:03:39.805624440Z INFO: INTERCONNECT MODE
[must-gather-lptfw] POD 2023-07-17T16:03:39.805624440Z INFO: Gathering ovn-kubernetes DBs
[must-gather-lptfw] POD 2023-07-17T16:03:39.923969317Z INFO: Gathering OVN_Northbound from ovnkube-node-564lr...
[must-gather-lptfw] POD 2023-07-17T16:03:39.924174701Z INFO: Gathering OVN_Northbound from ovnkube-node-8d8qp...
[must-gather-lptfw] POD 2023-07-17T16:03:39.924422506Z INFO: Gathering OVN_Northbound from ovnkube-node-qz4sq...
[must-gather-lptfw] POD 2023-07-17T16:03:39.924655881Z INFO: Gathering OVN_Northbound from ovnkube-node-r5srt...
[must-gather-lptfw] POD 2023-07-17T16:03:39.924895985Z INFO: Gathering OVN_Northbound from ovnkube-node-s4597...
[must-gather-lptfw] POD 2023-07-17T16:03:39.925120700Z INFO: Gathering OVN_Northbound from ovnkube-node-ssh2b...
[must-gather-lptfw] POD 2023-07-17T16:03:39.925373585Z INFO: Gathering OVN_Southbound from ovnkube-node-564lr...
[must-gather-lptfw] POD 2023-07-17T16:03:39.925601419Z INFO: Gathering OVN_Southbound from ovnkube-node-8d8qp...
[must-gather-lptfw] POD 2023-07-17T16:03:39.925824734Z INFO: Gathering OVN_Southbound from ovnkube-node-qz4sq...
[must-gather-lptfw] POD 2023-07-17T16:03:39.926047758Z INFO: Gathering OVN_Southbound from ovnkube-node-r5srt...
[must-gather-lptfw] POD 2023-07-17T16:03:39.926285043Z INFO: Gathering OVN_Southbound from ovnkube-node-s4597...
[must-gather-lptfw] POD 2023-07-17T16:03:39.926514077Z INFO: Gathering OVN_Southbound from ovnkube-node-ssh2b...
[must-gather-lptfw] POD 2023-07-17T16:03:40.279889738Z tar: Removing leading `/' from member names
[must-gather-lptfw] POD 2023-07-17T16:03:40.342438193Z tar: Removing leading `/' from member names
[must-gather-lptfw] POD 2023-07-17T16:03:40.439568926Z tar: Removing leading `/' from member names
[must-gather-lptfw] POD 2023-07-17T16:03:40.452053660Z tar: Removing leading `/' from member names
[must-gather-lptfw] POD 2023-07-17T16:03:40.514227418Z tar: Removing leading `/' from member names
[must-gather-lptfw] POD 2023-07-17T16:03:40.526194982Z tar: Removing leading `/' from member names
[must-gather-lptfw] POD 2023-07-17T16:03:40.538704778Z tar: Removing leading `/' from member names
[must-gather-lptfw] POD 2023-07-17T16:03:40.541750127Z tar: Removing leading `/' from member names
[must-gather-lptfw] POD 2023-07-17T16:03:40.551573349Z tar: Removing leading `/' from member names
[must-gather-lptfw] POD 2023-07-17T16:03:40.621253564Z tar: Removing leading `/' from member names
[must-gather-lptfw] POD 2023-07-17T16:03:40.625682981Z tar: Removing leading `/' from member names
[must-gather-lptfw] POD 2023-07-17T16:03:40.628531707Z tar: Removing leading `/' from member names
[must-gather-lptfw] POD 2023-07-17T16:03:41.729255505Z Wrote inspect data to must-gather.
[must-gather-lptfw] POD 2023-07-17T16:03:41.732211613Z INFO: Waiting for network log collection to complete ...
[must-gather-lptfw] POD 2023-07-17T16:03:41.732211613Z INFO: Waiting for ovnk database copies to complete ...
[must-gather-lptfw] POD 2023-07-17T16:03:41.732467948Z INFO: Copying ovnk databases complete.
[must-gather-lptfw] POD 2023-07-17T16:03:41.733288904Z 9.5M	must-gather/network_logs/ovnk_database_store
[must-gather-lptfw] POD 2023-07-17T16:03:41.734564339Z ovnk_database_store/
[must-gather-lptfw] POD 2023-07-17T16:03:41.734564339Z ovnk_database_store/ovnkube-node-564lr_sbdb
[must-gather-lptfw] POD 2023-07-17T16:03:41.750884809Z ovnk_database_store/ovnkube-node-qz4sq_nbdb
[must-gather-lptfw] POD 2023-07-17T16:03:41.755570111Z ovnk_database_store/ovnkube-node-8d8qp_nbdb
[must-gather-lptfw] POD 2023-07-17T16:03:41.764721990Z ovnk_database_store/ovnkube-node-ssh2b_sbdb
[must-gather-lptfw] POD 2023-07-17T16:03:41.787744461Z ovnk_database_store/ovnkube-node-qz4sq_sbdb
[must-gather-lptfw] POD 2023-07-17T16:03:41.803682663Z ovnk_database_store/ovnkube-node-r5srt_nbdb
[must-gather-lptfw] POD 2023-07-17T16:03:41.810852283Z ovnk_database_store/ovnkube-node-r5srt_sbdb
[must-gather-lptfw] POD 2023-07-17T16:03:41.827747554Z ovnk_database_store/ovnkube-node-ssh2b_nbdb
[must-gather-lptfw] POD 2023-07-17T16:03:41.835725661Z ovnk_database_store/ovnkube-node-564lr_nbdb
[must-gather-lptfw] POD 2023-07-17T16:03:41.842014904Z ovnk_database_store/ovnkube-node-8d8qp_sbdb
[must-gather-lptfw] POD 2023-07-17T16:03:41.865737498Z ovnk_database_store/ovnkube-node-s4597_nbdb
[must-gather-lptfw] POD 2023-07-17T16:03:41.874369007Z ovnk_database_store/ovnkube-node-s4597_sbdb
[must-gather-lptfw] POD 2023-07-17T16:03:41.897467720Z INFO: Network log collection complete.
```

### CODE REVIEW

File: collection-scripts/gather_network_logs
Comment (flavio-fernandes) [inline collection-scripts/gather_network_logs:79]: nice! I think this is good if we never go multi nodes per zone; which is the case.

**Review by flavio-fernandes — COMMENTED** [review]
/lgtm

**Review by RickJWagner — COMMENTED** [review]
Looks good.  PLM Approved.

**Comment by tssurya** [comment]
@RickJWagner : I see you are an approver for this repo, could you add the `approve` label as well or do I need to follow some specific process for merging the PR ?

File: collection-scripts/gather_network_logs
Comment (sferich888) [inline collection-scripts/gather_network_logs:23]: If we have 500 nodes do we really want data from all 500? Should this be limited in some way (or an extra script, where in a user specifies the nodes they want to collect this data from?

I worry this will take forever to collect on large clusters, and worry further that we will be collecting a large set of data no one will fully review.

**Review by sferich888 — CHANGES_REQUESTED** [review]

File: collection-scripts/gather_network_logs
Comment (tssurya) [reply collection-scripts/gather_network_logs:23]: > If we have 500 nodes do we really want data from all 500? Should this be limited in some way 

We could limit it to 100 nodes when using this main script.,perhaps?  Note that DB size on each node is going to be smaller than the one giant DB we collect today from the leader node because we are splitting the data across to each node locally rather than keeping it centralized.

> (or an extra script, where in a user specifies the nodes they want to collect this data from?

if we want to do extra script, that is fine - infact we could just opt-into `gather-dbs-from-all-nodes` script as add-on which is how it was before the 4.10 release

however the problem with capturing from specific DBs is that when users open bugs, they need to know which dbs to capture from and this is not trivial to determine, its possible pod moves between nodes and we might not know which node it was on when must-gather was triggered for collection. Since MG is a real-time event we mostly always need the full set of dbs across all nodes unless the user knows all the variables exactly (where was the pod, endpointslice, service) at the time if gathering.

> 
> I worry this will take forever to collect on large clusters, and worry further that we will be collecting a large set of data no one will fully review.

QQ: Does MG as of today actually scale all the way to 500 nodes? I usually see cases gathering sos-reports instead of a full must-gather from such large clusters... or inspect logs from specific namespaces usually...when the numbers of nodes is that high.

Comment (tssurya) [reply collection-scripts/gather_network_logs:23]: I will discuss with the wider networking team and come back to this.

Comment (tssurya) [reply collection-scripts/gather_network_logs:23]: > however the problem with capturing from specific DBs is that when users open bugs, they need to know which dbs to capture from and this is not trivial to determine, 

Let me clarify this more, pod is the simplest unit of repersentation and while it might be easy for us to say which ndoe the pod is on currently that is facing issues just before running a must-gather, what I meant was we have several dimentions of features in networking - egressIPs/egressFWs/network-policies/egressGWs which also have other associated hop routes etc configured on each node for network plumbing and hence depending on which feature we are debugging it might become crucial to gather all DBs.

I think as of now I see 3 options:
 
Option1 : remove DB collection as part of default MG and add a `gather-ovnk-dbs` script that will be opt-in -> this is always gather dbs from all nodes
Option2: do the above + take a node argument for specific node collections -> can be enhancement in future
Option3: Limit collecting DBs in the original script by default upto 100 nodes and then use gather-ovnk-dbs for large clusters -> this is confusing for users. I think we should stay consistent and go with option1.

Comment (tssurya) [reply collection-scripts/gather_network_logs:23]: We spoke about this in the team meeting. Everyone unitedly voted for collecting DBs by default and see if we can go that way before we need to pivot.

So here is what we are planning to do -> run the gathering in parallel across all nodes and tar them together as single entity instead of smaller tar's. 

@sferich888 : Do you have max limits for 500 node clusters - both time and size wise for a single MG tar file that I can use as values to compare a run against? That data can immensely help me versus running in blind with some numbers...

Comment (RickJWagner) [reply collection-scripts/gather_network_logs:23]: Hi @tssurya,
Eric (sferich888) is on PTO today.  If he doesn't have any further objections upon his return, I'll be glad to provide 'approve'.
I'll wait until he returns so I don't approve now and lose the opportunity for further refinement (if any exists).
Thanks,
Rick

File: collection-scripts/gather_network_logs
Comment (flavio-fernandes) [inline collection-scripts/gather_network_logs:36]: NIT: does PIDSDB have a limit? At some point, we will need to worry about how many nodes could be going at once, but that can be done at a different PR.

File: collection-scripts/gather_network_logs
Comment (ardaguclu) [inline collection-scripts/gather_network_logs:81]: Does this have an impact on CI jobs in terms of duration, size, etc.?. Is there any payload blocking job running this?. Probably you have discussed it but have you had measure the impact on must-gather output's size?.

I'm curious about this because we had a discussion recently about the must-gather pod's direct effect on other core components in master node due to the filling all the storage.

Comment (soltysh) [reply collection-scripts/gather_network_logs:81]: It does, we discussed this in length with @tssurya and @sferich888 yesterday, we agreed they will cut down the log gathering for networking nodes which is one of the biggest consumer (a followup work here or in oc), but will keep DBs, which will get looked at more in the future.

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

File: collection-scripts/gather_network_logs
Comment (tssurya) [reply collection-scripts/gather_network_logs:81]: yes what @soltysh said. SDN team is tracking this via https://issues.redhat.com/browse/SDN-3507. We will stop logging some of the heavy things were are logging in the container by default and instead divert those extra logs into a separate folder on the host that will not be gathered as part of default MG.

Comment (ardaguclu) [reply collection-scripts/gather_network_logs:81]: Sounds good to me, thanks @tssurya

**Pull request merged**
